dist/nginx.conf

```diff
@@ -18,8 +18,20 @@ server {
   listen [::]:80;
   server_name example.com;
   root /home/mastodon/live/public;
-  location /.well-known/acme-challenge/ { allow all; }
-  location / { return 301 https://$host$request_uri; }
+  # Hide /acme-challenge subdirectory and return 404 on all requests.
+  # It is somewhat more secure than letting Nginx return 403.
```

Comment:
Can you explain the claim that it is "somewhat more secure" to return a 404 on requests to `/acme-challenge` instead of 403?

I could only find this referenced in a couple of blog posts and the gist you linked, which is six years old.

Reply:
With a 403 response, the client may assume that the directory /.well-known/acme-challenge/ may be present on the server and used to obtain certificates, but it is not accessible. With a 404 response, it's already harder to know if this directory exists.